operators/src/upgrade/helm/client.rs

```diff
@@ -0,0 +1,249 @@
+        self.name = Some(name);
+    }
+
+    /// set a single option
+    pub fn set_opt(&mut self, key: String, value: String) {
+        self.opts.push((key, value));
+    }
+
+    /// reset array of options
+    pub fn set_opts(&mut self, options: Vec<(String, String)>) {
+        self.opts = options;
+    }
+
+    /// set namepsace
+    pub fn set_namespace(&mut self, ns: String) {
+        self.namespace = Some(ns);
+    }
+
+    /// set one value
+    pub fn set_values(&mut self, values: Vec<PathBuf>) {
+        self.values = values;
+    }
+
+    /// set one value
+    pub fn set_value(&mut self, value: PathBuf) {
+        self.values.push(value);
+    }
+
+    /// set a name
+    pub fn get_name(&self) -> &String {
+        self.name.as_ref().unwrap()
+    }
+
+    /// set a single option
+    pub fn get_opts(&self) -> &Vec<(String, String)> {
+        &self.opts
+    }
+
+    /// set namepsace
+    pub fn get_namespace(&self) -> &String {
+        self.namespace.as_ref().unwrap()
+    }
+
+    /// set one value
+    pub fn get_values(&self) -> &Vec<PathBuf> {
+        &self.values
+    }
+}
+
+/// chart representation
+#[derive(Deserialize, Debug, Clone, Default)]
+pub struct Chart {
+    name: String,
+    namespace: String,
+    revision: String,
+    updated: String,
+    status: String,
+    chart: String,
+    app_version: String,
+}
+
+impl Chart {
+    /// get name of the chart
+    pub fn name(&self) -> &String {
+        &self.name
+    }
+
+    /// get namespace
+    pub fn namespace(&self) -> &String {
+        &self.namespace
+    }
+
+    /// get installed chart by name s
+    pub fn get_installed_chart_by_name(
+        name: String,
+        namespace: String,
+    ) -> Result<Chart, HelmError> {
+        let exact_match = format!("^{}$", name);
+        println!("{}", exact_match);
+        let mut command = Command::new("helm");
+        command
+            .arg("ls")
+            .arg("--filter")
+            .arg(exact_match)
+            .arg("--namespace")
+            .arg(namespace)
+            .arg("--output")
+            .arg("json");
+
+        let output = command.output().map_err(|_| {
+            HelmError::HelmStdError("Error while running helm ls command".to_string())
+        })?;
+
+        let chart: Vec<Chart> = serde_json::from_slice(&output.stdout).map_err(|_| {
+            HelmError::SerdeDeserializationError("Deserialization error".to_string())
+        })?;
+
+        if chart.is_empty() {
+            return Err(HelmError::HelmChartNotFound(
+                "Helm chart not installed".to_string(),
+            ));
+        }
+
+        Ok(chart[0].clone())
+    }
+}
+
+/// Helm client
+#[derive(Debug, Clone)]
+pub struct HelmClient {
+    chart: Chart,
+    version: Option<String>,
+    args: HelmArgs,
+}
+
+impl HelmClient {
+    /// create a new helm client if helm is installed
+    pub fn new() -> Result<HelmClient, HelmError> {
+        let output = Command::new("helm")
```

Comment:
why not use a crate instead, such as https://github.com/stephenmuss/helm-api (haven't checked if it works or is appropriate, just a quick google search) ?

Reply:
This is for helm 2 and tiller components have been removed from helm 3.